deepchecks/nlp/checks/data_integrity/frequent_substrings.py

```diff
@@ -0,0 +1,227 @@
+                if ngram not in results:
+                    continue
+                if ngram in final_key:
+                    if ngram == final_key:
+                        continue
+                    ngram_freq = results[ngram]['freq']
+                    if ngram_freq - final_freq <= self.epsilon:
+                        del results[ngram]
+                    else:
+                        del results[final_key]
+                        final_key = ngram
+                        final_freq = ngram_freq
+        return results
+
+    def _get_significant_cut_ind(self, df):
+        significant_df = df[df['Frequency'] >= self.significant_threshold]
+        if len(significant_df) > 0:
+            return list(significant_df.index)[-1] + 1
+        return -1
+
+    def _filter_significant(self, df):
+        if len(df) == 1:
+            return df
+        significant_cut_ind = self._get_significant_cut_ind(df)
+        peak_cut_ind = self._get_peak_cut_ind(df)
+        return df[:max(significant_cut_ind, peak_cut_ind)].drop(columns=['diff'])
+
+    def run_logic(self, context: Context, dataset_kind):
+        """Run check.
+
+        Returns
+        -------
+        CheckResult
+            value is a df with 'Text', 'Number of Samples', 'Frequency' and 'Sample IDs' for each substring.
+
+        Raises
+        ------
+        DeepchecksValueError
+            If the Dataset is empty.
+        """
+        dataset = context.get_data_by_kind(dataset_kind).sample(self.n_samples, random_state=self.random_state)
+        dataset = t.cast(TextData, dataset)
+        if dataset.n_samples == 0:
+            raise DeepchecksValueError('Dataset cannot be empty')
+        data = list(zip(dataset.get_original_text_indexes(), dataset.text))
+
+        substrings_dict = self._get_frequent_ngrams(data)
+        substrings_dict = self._filter_longest_substrings_from_equal(substrings_dict)
+
+        if len(substrings_dict) == 0:
+            value = {}
+            display = None
+
+        else:
+            sorted_substrings = sorted(substrings_dict.items(), key=lambda x: (x[1]['freq'], x[0]), reverse=True)
+            df = pd.DataFrame({
+                'Text': [item[0] for item in sorted_substrings],
+                'Frequency': [item[1]['freq'] for item in sorted_substrings],
+                'Sample IDs': [item[1]['original_indexes'] for item in sorted_substrings]
+            })
+
+            df = self._filter_significant(df)
+            df['Number of Samples'] = df['Sample IDs'].str.len()
+
+            value = df
+            percent_of_frequent = sum(df['Number of Samples'])
+            display = [
+                f'{format_percent(percent_of_frequent)} of data samples share common substrings.',
+                'Each row in the table shows an example of a frequent substring and the number of times it appears.',
+                df[['Text', 'Number of Samples', 'Frequency']].iloc[slice(0, self.n_to_show)]
```

Comment:
For display, better display frequency in percents and change its name accordingly.